src/tasks/otomi/otomi-chart.ts

```diff
@@ -0,0 +1,84 @@
+import { omit, merge, pick } from 'lodash'
+import yaml from 'js-yaml'
+import fs from 'fs'
+import $RefParser from '@apidevtools/json-schema-ref-parser'
+
+import { cleanEnv, OTOMI_VALUES_INPUT, OTOMI_SCHEMA_PATH, OTOMI_VALUES_TARGET } from '../../validators'
+
+const env = cleanEnv({
+  OTOMI_VALUES_INPUT,
+  OTOMI_SCHEMA_PATH,
+  OTOMI_VALUES_TARGET,
+})
+
+const schemaKeywords = ['properties', 'anyOf', 'allOf', 'oneOf']
+
+export default function extractSecrets(schema: any, parentKey: string): Array<string> {
+  return Object.keys(schema)
+    .flatMap((key) => {
+      const childObj = schema[key]
+      if (typeof childObj !== 'object') return false
+      if ('x-secret' in childObj) return `${parentKey}.${key}`
+      if (key in ['anyOf', 'allOf', 'oneOf']) return extractSecrets(childObj, parentKey)
+      const address = schemaKeywords.includes(key) ? parentKey : `${parentKey}.${key}`
+      return extractSecrets(childObj, address)
+    })
+    .filter(Boolean) as Array<string>
+}
+
+function mergeValues(targetPath: string, valueObject): void {
+  let bsValues
+  if (fs.existsSync(targetPath)) {
+    bsValues = yaml.load(fs.readFileSync(targetPath).toString())
+  }
+  if (!bsValues) {
+    bsValues = {}
+  }
+  merge(bsValues, valueObject)
+  fs.writeFileSync(targetPath, yaml.safeDump(bsValues))
+}
+
+async function main(): Promise<void> {
+  const values = yaml.safeLoad(fs.readFileSync(env.OTOMI_VALUES_INPUT, 'utf8')) as any
+
+  // creating secret files
+  const schema = yaml.safeLoad(fs.readFileSync(env.OTOMI_SCHEMA_PATH, 'utf8')) as any
+  const derefSchema = await $RefParser.dereference(schema)
+  const cleanSchema = omit(derefSchema, ['definitions', 'properties.teamConfig']) // FIXME: lets fix the team part later
+  const secretsJsonPath = extractSecrets(cleanSchema, 'root').map((str) => str.replace('root.', ''))
```

Comment:
why the need to provide 'root'? Just write the function to make that param optional...